src/app/_components/Search/Search.tsx

```diff
@@ -497,42 +535,91 @@ function SearchInput({ searchTermFromQueryParams = '' }) {
   const searchTerm = useAppSelector(selectSearchTerm);
   const searchResponse = useSearchQuery(searchTerm, true);
   const isLoading = searchResponse.isLoading;
+  const isMobile = useBreakpointValue({ base: true, lg: false });
+
+  useEffect(() => {
+    if (isMobile && isSearchFieldFocused) {
+      document.body.style.overflow = 'hidden';
+    } else {
+      document.body.style.overflow = 'auto';
+    }
+  }, [isMobile, isSearchFieldFocused]);
+
+  const Wrapper = useCallback(
+    function Wrapper({ children }: { children: ReactNode }) {
+      if (fullScreen) {
+        return (
+          <Flex
+            position={isSearchFieldFocused ? 'absolute' : undefined}
+            top={4}
+            left={3}
+            right={3}
+            zIndex={2}
+            background={'surfaceTertiary'}
+          >
+            {children}
+          </Flex>
+        );
+      }
+      return children;
+    },
+    [isSearchFieldFocused, fullScreen]
+  );
 
   return (
-    <HStack width="full" position={'relative'} zIndex={1}>
-      <SearchInput searchTermFromQueryParams={searchTermFromQueryParams} />
-      {isSearchFieldFocused && (
-        <SearchResultsWrapper>
-          {isLoading ? null : searchResponse && searchResponse.data ? (
-            <Stack pt={18}>
-              <SearchResults
-                recentResults={[searchResponse.data]}
-                iconType={'enter'}
-                px={3}
-                pb={4}
-              />
-            </Stack>
-          ) : (
-            <>
-              <Stack gap={6}>
-                <Stack px={4} gap={3} pt={16}>
-                  <KeywordsPreview />
-                </Stack>
-                <SearchResults title={'Recent results'} recentResults={recentResults} />
+    <Wrapper>
+      <HStack width="full" position={'relative'} zIndex={'modal'}>
+        <SearchInput searchTermFromQueryParams={searchTermFromQueryParams} />
+        {isSearchFieldFocused && (
+          <SearchResultsWrapper height={fullScreen ? '100vh' : 'auto'}>
+            {isLoading ? null : searchResponse && searchResponse.data ? (
+              <Stack pt={18}>
+                <SearchResults
+                  recentResults={[searchResponse.data]}
+                  iconType={'enter'}
+                  px={3}
+                  pb={4}
+                />
               </Stack>
-              <QuickLinks />
-            </>
-          )}
-        </SearchResultsWrapper>
+            ) : (
+              <>
+                <Stack gap={6}>
+                  <Stack px={4} gap={3} pt={16}>
+                    <KeywordsPreview />
+                  </Stack>
+                  <SearchResults title={'Recent results'} recentResults={recentResults} />
+                </Stack>
+                {!fullScreen && <QuickLinks />}
+              </>
+            )}
+          </SearchResultsWrapper>
+        )}
+      </HStack>
+      {isSearchFieldFocused && (
```

Comment:
The implementation might be more straightforward using Chakra UI portals, which we use for modals. Did you look into if that would work?

Reply:
can you elaborate on the value of adding portal vs just conditionally render the component in this instance?